Gems/Atom/Asset/ImageProcessingAtom/Code/Source/ImageLoader/TIFFLoader.cpp

```diff
@@ -137,389 +154,303 @@ namespace ImageProcessingAtom
-            uint32_t dwPhotometric = 0;
-            TIFFGetField(tif, TIFFTAG_SAMPLESPERPIXEL, &dwChannels);
-            TIFFGetField(tif, TIFFTAG_PHOTOMETRIC, &dwPhotometric);
-
-            uint32_t dwWidth = 0;
-            uint32_t dwHeight = 0;
-            TIFFGetField(tif, TIFFTAG_IMAGEWIDTH, &dwWidth);
-            TIFFGetField(tif, TIFFTAG_IMAGELENGTH, &dwHeight);
-
-            EPixelFormat eFormat = ePixelFormat_R16G16B16A16F;
-
-            if (dwChannels == 1 && dwPhotometric != PHOTOMETRIC_MINISBLACK)
+            // Copy one channel of one pixel from source to destination, and optionally invert the value.
+            auto CopyPixelChannel = [bitsPerChannel, pixelValueScale, &buf, &dst, &minChannelValue, &maxChannelValue]
+                (uint32_t destPixelChannelIndex, uint32_t srcPixelChannelIndex, bool invert = false)
             {
-                eFormat = ePixelFormat_R16F;
-            }
-
-            IImageObject* pRet = IImageObject::CreateImage(dwWidth, dwHeight, 1, eFormat);
+                if (bitsPerChannel == 8)
+                {
+                    dst[destPixelChannelIndex] = invert ? (0xFF - buf[srcPixelChannelIndex]) : buf[srcPixelChannelIndex];
+                }
+                else if (bitsPerChannel == 16)
+                {
+                    // Alias the base pointers to a 16-bit channel type, then use the channel index to get to the correct pixel & channel
+                    uint16_t* buf16 = reinterpret_cast<uint16_t*>(buf.data());
+                    uint16_t* dst16 = reinterpret_cast<uint16_t*>(dst);
+                    dst16[destPixelChannelIndex] = invert ? (0xFFFF - buf16[srcPixelChannelIndex]) : buf16[srcPixelChannelIndex];
+                }
+                else
+                {
+                    // Alias the base pointers to a 32-bit channel type, then use the channel index to get to the correct pixel & channel
+                    float* buf32 = reinterpret_cast<float*>(buf.data());
+                    float* dst32 = reinterpret_cast<float*>(dst);
 
-            uint8_t* dst;
-            uint32_t dwPitch;
-            pRet->GetImagePointer(0, dst, dwPitch);
+                    // GeoTIFFs might have a pixel scale, so apply it.
+                    const float scaledValue = buf32[srcPixelChannelIndex] * pixelValueScale;
 
-            AZStd::vector<char> buf(dwPitch);
+                    // Track min/max values, but exclude the lowest float value, as that might be a "no data" value for GeoTIFFs.
+                    if (scaledValue > AZStd::numeric_limits<float>::lowest())
+                    {
+                        minChannelValue = AZStd::min(minChannelValue, scaledValue);
+                        maxChannelValue = AZStd::max(maxChannelValue, scaledValue);
+                    }
 
-            static const uint16_t zero = 0;
-            static const uint16_t one = 1;
+                    dst32[destPixelChannelIndex] = invert ? (1.0f - scaledValue) : scaledValue;
```

Comment:
nit: You can removed the invert check since HDR input won't have invert anyway.

Reply:
I'll remove and just add an error message for any photometric separated files that try to use 16-bit or 32-bit floats.